internal/graph/check_test.go

```diff
@@ -2998,3 +3000,338 @@ func TestIteratorReadStartingFromUser(t *testing.T) {
+			expectedErr: condition.NewEvaluationError("x_y",
+				fmt.Errorf("tuple 'document:1#can_view@user:maria' is missing context parameters '[y]'")),
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			ts, err := typesystem.NewAndValidate(context.Background(), tt.model)
+			require.NoError(t, err)
+
+			contextStruct, err := structpb.NewStruct(tt.context)
+			require.NoError(t, err)
+
+			iterFunc := buildTupleKeyConditionFilter(context.Background(), contextStruct, ts)
+			result, err := iterFunc(tt.tupleKey)
+			if tt.expectedErr != nil {
+				require.Equal(t, tt.expectedErr.Error(), err.Error())
+			} else {
+				require.NoError(t, err)
+			}
+			require.Equal(t, tt.conditionMet, result)
+		})
+	}
+}
+
+func TestBuildCheckAssociatedObjects(t *testing.T) {
+	tests := []struct {
+		name                         string
+		model                        *openfgav1.AuthorizationModel
+		tuples                       []*openfgav1.TupleKey
+		context                      map[string]interface{}
+		dsError                      error
+		objectIDs                    []string
+		expectedError                bool
+		expectedResolveCheckResponse *ResolveCheckResponse
+	}{
+		{
+			name: "empty_iterator",
+			model: parser.MustTransformDSLToProto(`
+				model
+					schema 1.1
+
+				type user
+				type group
+					relations
+						define member: [user]
+				type document
+					relations
+						define viewer: [group#member]`),
+			tuples:  []*openfgav1.TupleKey{},
+			dsError: nil,
+			objectIDs: []string{
+				"2", "3",
+			},
+			context: map[string]interface{}{},
+			expectedResolveCheckResponse: &ResolveCheckResponse{
+				Allowed: false,
+				ResolutionMetadata: &ResolveCheckResponseMetadata{
+					DatastoreQueryCount: 1,
+				},
+			},
+			expectedError: false,
+		},
+		{
+			name: "bad_ds_call",
+			model: parser.MustTransformDSLToProto(`
+				model
+					schema 1.1
+
+				type user
+				type group
+					relations
+						define member: [user]
+				type document
+					relations
+						define viewer: [group#member]`),
+			tuples:  []*openfgav1.TupleKey{},
+			dsError: fmt.Errorf("bad_ds_call"),
+			objectIDs: []string{
+				"2", "3",
+			},
+			context:                      map[string]interface{}{},
+			expectedResolveCheckResponse: nil,
+			expectedError:                true,
+		},
+		{
+			name: "non_empty_iterator_match_objectIDs",
+			model: parser.MustTransformDSLToProto(`
+				model
+					schema 1.1
+
+				type user
+				type group
+					relations
+						define member: [user]
+				type document
+					relations
+						define viewer: [group#member]`),
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("group:1", "member", "user:maria"),
+				tuple.NewTupleKey("group:2", "member", "user:maria"),
+				tuple.NewTupleKey("group:3", "member", "user:maria"),
+				tuple.NewTupleKey("group:4", "member", "user:maria"),
+			},
+			dsError: nil,
+			objectIDs: []string{
+				"2", "3",
+			},
```

Comment:
Nitpick: you don't need 4 tuples to create a truthy response, just one tuple should be enough. Or are we also testing the behavior of `tupleIDInSortedSet`, which already has its own tests? 😄

Reply:
Will simplify the test - tupleIDInSortedSet already has its own tests